react-forms/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,38 @@
+import { FormEvent, useState } from 'react';
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    console.log({ username, password });
+  }
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <label htmlFor="username">
+            Username:{' '}
+            <input
+              name="username"
+              value={username}
+              onChange={(e) => setUsername(e.target.value)}></input>
+          </label>
+        </div>
+        <div>
+          <label htmlFor="password">
```

Comment:
When you use the htmlFor attribute you no longer have to wrap your input in a label.
But you do have to give the input element an id that matches the htmlFor attribute.